src/components/dls/ModalNew/Modal.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import styles from './Modal.module.scss';
+
+const Modal = ({ children, trigger }) => (
+  <DialogPrimitive.Root>
+    <DialogPrimitive.Overlay className={styles.overlay} />
+    {trigger}
+    <Content>{children}</Content>
+  </DialogPrimitive.Root>
+);
+
+const Content = React.forwardRef<HTMLDivElement, any>(({ children, ...props }, forwardedRef) => (
+  <DialogPrimitive.Content {...props} ref={forwardedRef} className={styles.content}>
+    {children}
+  </DialogPrimitive.Content>
+));
+
+const Body = ({ children }) => <div className={styles.body}>{children}</div>;
```

Comment:
I think we should split the body into header and the body itself. So we add another section called `Modal.Header` that will include `Title` and `Subtitle` and the body will only include the content itself.

Reply:
i see, agree